sync-android/src/test/java/com/cloudant/sync/datastore/encryption/CacheKeyProviderTests.java

```diff
@@ -0,0 +1,121 @@
+package com.cloudant.sync.datastore.encryption;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeNotNull;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+public class CacheKeyProviderTests {
+    @Before
+    public void beforeMethod() {
+        assumeNotNull(ProviderTestUtil.getContext());
+    }
+
+    @Test
+    public void testCreateProviderWithIdentifier() {
+        CacheKeyProvider provider = new CacheKeyProvider(new AndroidKeyProvider(ProviderTestUtil
+                .getContext(),
+                ProviderTestUtil.password, ProviderTestUtil.getUniqueIdentifier()));
+
+        // This will cause the provider to generate a key using the AndroidKeyProvider. The key
+        // will be encrypted, persisted, and stored in memory.
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        // This will cause the provider to load the cached key.
+        EncryptionKey cachedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not cached", cachedKey);
+
+        // Assert cached key and created key are the same
+        assertTrue("cachedKey did not match createdKey", Arrays.equals(createdKey.getKey(),
+                cachedKey.getKey()));
+
+        AndroidKeyProvider androidKeyProvider = (AndroidKeyProvider) provider.getKeyProvider();
```

Comment:
Using the AndroidKeyProvider the EncryptionKey is stored on device. It's cleared to put the device back into a clean state for each test.